routing/router_test.go

```diff
@@ -4130,28 +4138,25 @@ func TestSendMPPaymentFailedWithShardsInFlight(t *testing.T) {
 	var failedAttempt channeldb.HTLCAttempt
 	controlTower.On("FailAttempt",
 		identifier, mock.Anything, mock.Anything,
-	).Return(&failedAttempt, nil).Run(func(args mock.Arguments) {
-		// Whenever this method is invoked, we will mark the first
-		// active attempt as failed and exit.
-		failedAttempt = payment.HTLCs[0]
-		failedAttempt.Failure = &channeldb.HTLCFailInfo{}
-		payment.HTLCs[0] = failedAttempt
-	})
+	).Return(&failedAttempt, nil)
```

Comment:
Q: I was wondering why we return a pointer to an empty HTLCAttempt here in the new mockControlTower struct?

Reply:
hmm not sure i understand the question...